packages/openapi/src/openapi-request-builder.ts

```diff
@@ -164,23 +164,66 @@ export class OpenApiRequestBuilder<ResponseT = any> {
 
   private getPath(): string {
     const pathParameters = this.parameters?.pathParameters || {};
+    const path = this.pathPattern;
+
+    // Get the innermost curly bracket pairs with non-empty and legal content as placeholders.
+    const placeholders = path.match(/{[^/?#{}]+}/g);
+    if (!placeholders) {
```

Comment:
Please correct me if I'm wrong, but the placeholder names are unique, correct? That is how you should be able to match them exactly without sorting them. As these checks have been done in the generator there is no need to do them on request building (except the one checking for invalid values).

With this your code could be much simpler

- for every placeholder
  - get the path parameter by name
  - check that the value is valid
  - replace it in the path
- return the resulting path

Reply:
Done. 
Since we can assume that users will always use our `openapi-generator`, there is no need to add checks anymore.